spec/unit/provider/postgresql_conf/ruby_spec.rb

```diff
@@ -11,6 +11,7 @@
     allow(provider).to receive(:file_path).and_return('/tmp/foo')
     allow(provider).to receive(:read_file).and_return('foo = bar')
     allow(provider).to receive(:write_file).and_return(true)
+    allow(provider).to receive(:resource).and_return(key: 'your_key', line_number: 1, value: 'foo')
```

Comment:
I meant to change line 7 to:
```ruby
let(:resource) { Puppet::Type.type(:postgresql_conf).new(name: 'foo', key: 'foo', value: 'bar') }
```

Reply:
ah okay. will update

Reply:
mhm this line is still requires to get the tests passing

Reply:
This confuses me and now I wonder what's wrong. Perhaps tomorrow I'll have some idea.

Reply:
So I think I know what is going on. Line 7 instantiates a new Puppet type with the desired state. The provider has the actual state. You're mocking that here. So by mocking this, you know it is already in place.

So at the very least I'd expect you to only mock it when you want it to exist, not before each test.

It also makes me wonder about something else. `key` is a parameter, not a property. So I'm not sure it should be loaded from the provider.

Reply:
This line is now redundant, right?
```suggestion
```